webapp/src/pages/RegisterPage/RegisterPage.tsx

```diff
@@ -0,0 +1,161 @@
+import { useState } from 'react';
+import { useNavigate } from 'react-router-dom';
+import {
+  Button,
+  ButtonVariant,
+  Input,
+  InputVariant,
+  Text,
+  TextColor,
+  TextVariant,
+  TextWeight,
+} from '../../shared/components';
+import { ResponseError } from '../../shared/generated';
+import { authApi } from '../../shared/services/ApiService';
+import { SubmitStatus } from '../../shared/types';
+import { LOGIN_OPTIONS_URL } from '../../shared/urls';
+import './RegisterPage.css';
+
+export default function RegisterPage() {
+  const [username, setUsername] = useState('');
+  const [fullName, SetFullName] = useState('');
+  const [email, setEmail] = useState('');
+  const [password, setPassword] = useState('');
+  const [confirmationPassword, setConfirmationPassword] = useState('');
+  const [status, setStatus] = useState<SubmitStatus>({
+    type: 'pending',
+    message: '',
+  });
+  const navigate = useNavigate();
+  async function register() {
+    try {
+      await authApi.authControllerRegisterLocalUser({
+        registerUserDto: {
+          username,
+          password,
+          confirmationPassword,
+          email,
+          fullName,
+        },
+      });
+      setStatus({
+        type: 'success',
+        message: 'Registration complete',
+      });
+      setTimeout(() => {
+        navigate(LOGIN_OPTIONS_URL);
+      }, 2000);
+    } catch (error) {
+      if (error instanceof ResponseError) {
+        if (error.response.status === 422) {
+          setStatus({
+            type: 'error',
+            message: 'Username already registered',
+          });
+        } else if (error.response.status === 400) {
+          setStatus({
+            type: 'error',
+            message: 'Invalid email or wrong passwords',
+          });
+        } else {
+          setStatus({ type: 'error', message: `${error.response.statusText}` });
+        }
+      } else if (error instanceof Error) {
+        setStatus({ type: 'error', message: `${error.message}` });
+      }
+    }
+  }
+  const handleOnSubmit = (e: React.FormEvent<HTMLFormElement>): void => {
+    e.preventDefault();
+    register();
```

Comment:
```
Promise returned from register is ignored 
```
couldn't we attach a `.catch((e)=>console.error(e));` to `register()` ?

Reply:
Yes, I promise I thought I did it, maybe I didn't push. I add it